src/Illuminate/Cache/Console/ClearCommand.php

```diff
@@ -46,13 +47,33 @@ public function __construct(CacheManager $cache)
 	 */
 	public function fire()
 	{
-		$this->laravel['events']->fire('cache:clearing');
+		$storeName = $this->argument('store');
 
-		$this->cache->flush();
+		try {
+			$cacheStore = $this->cache->store($storeName);
+		} catch (\InvalidArgumentException $e) {
```

Comment:
For example, we throw an exception when a command is not found, and that's totally readable.